### PR TITLE
Point probes

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
 Extends the custom outputs with point probes (nearest point, no interpolation is done).

## Related Work
#1907 cc @DonalMcC

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [X] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

**Comment by bigfooted** [comment]
Nice! 

Since the probe location is taken from the nearest neighbor, should the actual probe location also be written to output?

File: SU2_CFD/src/output/CFlowOutput.cpp [hunk 1]
```diff
@@ -765,16 +765,37 @@
...
+          const su2double dist = GeometryToolbox::SquaredDistance(nDim, coord, geometry->nodes->GetCoord(iPoint));
+          if (dist < minDist) {
+            minDist = dist;
+            minPoint = iPoint;
           }
         }
+        /*--- Decide which rank owns the probe. ---*/
+        su2double globMinDist;
+        SU2_MPI::Allreduce(&minDist, &globMinDist, 1, MPI_DOUBLE, MPI_MIN, SU2_MPI::GetComm());
+        output.iPoint = (minDist == globMinDist) ? minPoint : CustomOutput::PROBE_NOT_OWNED;
```
[the remark this answers is not in the document]
Comment (bigfooted) [reply SU2_CFD/src/output/CFlowOutput.cpp:798]: Should this be |minDist-globMinDist| < epsilon ?

[the remark this answers is not in the document]
Comment (pcarruscag) [reply SU2_CFD/src/output/CFlowOutput.cpp:798]: I can write it like that but it is safe to compare the results of min/max with ==

**Comment by pcarruscag** [comment]
I can write the probe location for debugging purposes.
The probes have access to the same things as the custom outputs and they only output one scalar.
So if someone wants to probe the entire flow field they need to define one probe per variable.

**Pull request merged**